threejs/fxhash_Rooms/src/index.js

```diff
@@ -57,15 +59,82 @@ material.roughness = 0.7;
 gui.add(material, "metalness").min(0).max(1).step(0.001);
 gui.add(material, "roughness").min(0).max(1).step(0.001);
 
+/**
+ * Helpers
+ * 
+ * Description: Axes and grid helpers
+ * TODO: Remove before publishing
+*/
+
+// Axes Helper 
+const axes = new THREE.AxesHelper( 5 );
+scene.add( axes );
+
+// Grid Helper
+const gridSize = 10;
+const gridDivisions = 10;
+const gridHelper = new THREE.GridHelper( gridSize, gridDivisions, 'black', 'pink' );
+scene.add( gridHelper );
+
 /**
  * Objects
- *
- * Object meshes are generated by the anaverse-compatible
- * figures.js script. This script yields a "figures" array
- * containing the information necessary to build & position
- * a THREE mesh.
- *
- */
+*/
+
+// Simple Closed Polygon Algorithm // Source: https://openprocessing.org/sketch/1626897 by Michael Hoehn
+let resolution = Math.floor(4 + fxrand() * 11); 
+let stepSize = 1 + fxrand() * 4; 
+let radius = Math.floor(2 + fxrand() * 5); 
+let x = [];
+let y = []; 
+let angle = (Math.PI / 180) * (360 / resolution);
+const positions = [];
+
+for ( let i = 0; i < resolution; i++ ) {
+  x.push(Math.cos(angle * i) * radius);
+  y.push(Math.sin(angle * i) * radius);
+}
+
+for ( let i = 0; i < resolution; i++ ) {
+  const splinePts = new THREE.Vector2(
+    x[i] += fxrand() * stepSize - (stepSize/2),
+    y[i] += fxrand() * stepSize - (stepSize/2)
+  )
+  positions.push(splinePts);
+}
+
+const roomShape = new THREE.Shape();
+const extrudeSettings = {
+	steps: 1,
+	depth: 0.2,
+	bevelEnabled: false,
+	bevelThickness: 0,
+	bevelSize: 0,
+	bevelOffset: 0,
+	bevelSegments: 0
+};
+roomShape.setFromPoints( positions );
+
+const roomExtrusionGeometry = new THREE.ExtrudeGeometry( roomShape, extrudeSettings );
+const roomMaterial = new THREE.MeshBasicMaterial( { color: 'gray', wireframe: false } );
+const roomMesh = new THREE.Mesh( roomExtrusionGeometry, material );
+roomMesh.castShadow = true;
+roomMesh.receiveShadow = true;
+
+// Create the room outline using positions[]
+const splineGeometry = new THREE.BufferGeometry().setFromPoints(positions);
+const splineMaterial = new THREE.LineBasicMaterial( { color: 'black' } ); 
+const splineObject = new THREE.Line( splineGeometry, splineMaterial );
+
+scene.add(splineObject);
+scene.add(roomMesh);
+
+splineObject.rotation.x = -Math.PI * 0.5;
+roomMesh.rotation.x = -Math.PI * 0.5;
```

Comment:
@Anaglyph-ic as discussed, I've introduced a logical array of ordered vertices within the `positions[]` array. I'm using that here to first created the `roomShape` and subsequently the `roomExtrusionGeometry` and `roomMesh`. 

I'm liking the random forms created [here](https://github.com/michaelhoehn/fxHash-collabs/pull/3#issuecomment-1215678776) and ideally would want to include them in the overall model. I'm aware this may be a problem correct? Would you recommend only using the walls (which is my next step) and letting the floor be a ground plane instead of the exact mesh extrusion shape? Thanks!

Reply:
By only using the walls, I can ensure we are only using Box geometries which are fully compliant with your api. 👍 

If using the ground plane is preferrable, I can simply size the plane according to a bounding box of the initial curve used to create the perimeter. So that shouldn't be a problem.